lib/Optimizer/Transforms/FactorQuantumAlloc.cpp

```diff
@@ -96,9 +96,8 @@ class FactorQuantumAllocationsPass
   using FactorQuantumAllocationsBase::FactorQuantumAllocationsBase;
 
   void runOnOperation() override {
-    func::FuncOp func = getOperation();
     LLVM_DEBUG(llvm::dbgs() << "Function before factoring quake alloca:\n"
-                            << func << "\n\n");
```

Comment:
The original code causes a compilation failure if the pass is ever changed to not run on a FuncOp. This change removes that sanity check.